File: Data Structures & Algorithms/is-anagram/submission-4.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) return false;
        std::vector<int> values(26, 0);
        for (int i = 0; i < s.length(); i++)
        {
            values[s[i] - 'a']++;
            values[t[i] - 'a']--;
        }

        for (auto value : values)
        {
            if (value != 0)
            {
                return false;
            }
        }

        return true;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        r = 1
        profit = 0

        while l < len(prices) and r < len(prices):
            if (prices[l] < prices[r]):
                profit = max(profit, prices[r] - prices[l])
            if (prices[r] < prices[l]):
                l = r
            r += 1

        return profit